backend/infrahub/git/base.py

```diff
@@ -864,37 +917,41 @@ def check_connectivity(cls, name: str, url: str) -> None:
         except GitCommandError as exc:
             cls._raise_enriched_error_static(name=name, location=url, error=exc)
 
-    def _raise_enriched_error(self, error: GitCommandError, branch_name: str | None = None) -> NoReturn:
-        self._raise_enriched_error_static(
-            error=error, name=self.name, location=self.location, branch_name=branch_name or self.default_branch
-        )
+    async def _raise_enriched_error(self, error: GitCommandError, branch_name: str | None = None) -> NoReturn:
+        try:
+            self._raise_enriched_error_static(
+                error=error, name=self.name, location=self.location, branch_name=branch_name or self.default_branch
+            )
+        except RepositoryError as exc:
+            await self._update_operational_status(
+                status={
+                    RepositoryConnectionError: RepositoryOperationalStatus.ERROR_CONNECTION,
+                    RepositoryCredentialsError: RepositoryOperationalStatus.ERROR_CRED,
+                }.get(type(exc), RepositoryOperationalStatus.ERROR)
```

Comment:
I think we could move the logic of the type of error to the error itself. So if `self._update_operational_status` takes an optional error parameter we could have an attribute of that error that would indicate the type of `RepositoryOperationalStatus` status that it belongs to. Then the code outside of the error. doesn't have to know what the error would indicate. I.e if RepositoryError had an "OPERATIONAL_STATUS" attribute that defaulted to `RepositoryOperationalStatus.ERROR` than the child error classes could override this value and use something more specific (for now I don't know if this might lead to circular imports as we still have constants under infrahub/core. If that's the case we can perhaps create a follow up issue for this.

Reply:
Yes, I was aiming for that in the first place but ended up in, as you guessed, a circular import issue, hence the not so pretty inline dict lookup I came up with.